spanner/read.go

```diff
@@ -677,6 +680,10 @@ func (d *resumableStreamDecoder) tryRecv(mt *builtinMetricsTracer, retryer gax.R
 	if d.err == nil {
 		d.q.push(res)
 		if res.GetLast() {
+			if span := otrace.SpanFromContext(d.stream.Context()); span != nil && span.IsRecording() {
+				span.SetStatus(otcodes.Ok, "Stream finished successfully")
+				span.End()
+			}
 			go func(s streamingReceiver) {
```

Comment:
Would this change then also make it possible to move the `d.cancel()` call from the goroutine into the main thread (and remove the extra s.Recv() call)? Or would that then still cause either the trace or the call server-side to show up as cancelled?

Reply:
Removed the extra go routine

Reply:
iter.Stop will cancel the routine now anyways